casper/src/main/scala/coop/rchain/casper/ReportHotStore.scala

```diff
@@ -0,0 +1,114 @@
+package coop.rchain.casper
+
+import scodec.bits.BitVector
+import cats.implicits._
+import cats.effect.Sync
+import com.google.protobuf.ByteString
+import coop.rchain.metrics.Span
+import coop.rchain.scodec.codecs.seqOfN
+import scodec.codecs.{bool, bytes, int32, int64, uint8, variableSizeBytesLong}
+import scala.collection.concurrent.TrieMap
+import scodec.Codec
+import scodec.codecs.{discriminated, uint2}
+import coop.rchain.rspace.ReportingRspace._
+import coop.rchain.shared.AttemptOpsF.RichAttempt
+import coop.rchain.shared.Serialize
+import coop.rchain.scodec.codecs.SeqCodec
+
+/**
+  This is a temporary solution for the performance problem which is caused by the
+  reporting api. Exchanges highly rely on this api to get transactions info.But
+  every time request to the event data, the node would replay the block once again
+  which makes the api so slow and can not processed concurrently.
+
+  The ReportHotStore is a in memory store for event data in every block.
+  */
+trait ReportHotStore[F[_]] {
```

Comment:
The name HotStore can be confusing because HotStore in RSpace is used to collect changes for event log (hot store actions).

Reply:
@zsluedem might be ReportingCache?